### PR TITLE
UIDATIMP-558 Add hover text for "Cannot be mapped" icon in field mapp…

### PR DESCRIPTION
**Pull request by IvanKryzhanovskyi**
## Purpose
To add hover text to the "cannot be mapped" icon, to help distinguish from the icon for not mapped

## Approach
1. Extend ProhibitionIcon component and getUnmappableValueById util function with receiving fieldName prop
2. Update all necessary calls of those functions.

## Tickets
[UIDATIMP-558](https://issues.folio.org/browse/UIDATIMP-558)

## Screenshot
tooltip

### CODE REVIEW

File: src/components/ProhibitionIcon/ProhibitionIcon.js [hunk 1]
```diff
@@ -2,29 +2,47 @@ import React from 'react';
...
 /**
  * Component for rendering the circle backslash character with given message as an aria-label
  *
  * @param {string} [ariaLabel] - A label or a translation id for the aria-label attribute
+ * @param {string} fieldName - field name for creation of required tooltip id attribute
  * @returns {{}}
  */
-export const ProhibitionIcon = ({ ariaLabel }) => {
+export const ProhibitionIcon = ({
+  ariaLabel, fieldName,
```
Comment (mariia-aloshyna) [inline src/components/ProhibitionIcon/ProhibitionIcon.js:17]:
```suggestion
  ariaLabel,
  fieldName,
```

File: src/components/ProhibitionIcon/ProhibitionIcon.js [hunk 2]
```diff
@@ -2,29 +2,47 @@ import React from 'react';
...
-    >
-      &#8416;
-    </span>
+    <div className={css.iconContainer}>
+      <Tooltip
+        id={fieldName}
+        text={message || <FormattedMessage id="ui-data-import.settings.mappingProfiles.canNotBeMapped" />}
+      >
+        {({
+          ref, ariaIds,
```
Comment (mariia-aloshyna) [inline src/components/ProhibitionIcon/ProhibitionIcon.js:30]:
```suggestion
          ref,
          ariaIds,
```

File: src/components/ProhibitionIcon/ProhibitionIcon.js [hunk 3]
```diff
@@ -2,29 +2,47 @@ import React from 'react';
...
+ * @param {string} fieldName - field name for creation of required tooltip id attribute
  * @returns {{}}
  */
-export const ProhibitionIcon = ({ ariaLabel }) => {
+export const ProhibitionIcon = ({
+  ariaLabel, fieldName,
+}) => {
   const ariaLabelIsTranslationId = ariaLabel && ariaLabel.includes('ui-');
   const ariaLabelIsMessageString = ariaLabel && !ariaLabel.includes('ui-');
-  const translationId = ariaLabelIsTranslationId ? ariaLabel : 'ui-data-import.noValueAllowed';
```
Comment (mariia-aloshyna) [inline src/components/ProhibitionIcon/ProhibitionIcon.js:16]: Please don't get rid of this translation - screen readers should read the current icon as "No value allowed" (https://issues.folio.org/browse/UIDATIMP-471?focusedCommentId=74991&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-74991)
I'd suggest creating two separate messages - for `aria-label` and `Tooltip text`

File: src/components/ProhibitionIcon/ProhibitionIcon.js [hunk 4]
```diff
@@ -42,4 +60,7 @@ export const ProhibitionIcon = ({ ariaLabel }) => {
   return renderIcon();
 };
 
-ProhibitionIcon.propTypes = { ariaLabel: PropTypes.string };
+ProhibitionIcon.propTypes = {
+  ariaLabel: PropTypes.string,
+  fieldName: PropTypes.string,
```
Comment (mariia-aloshyna) [inline src/components/ProhibitionIcon/ProhibitionIcon.js:65]: Looks like this prop is required, isn't it?

File: src/components/ProhibitionIcon/ProhibitionIcon.js (hunk 1 above)
Comment (IvanKryzhanovskyi) [reply src/components/ProhibitionIcon/ProhibitionIcon.js]: Done

File: src/components/ProhibitionIcon/ProhibitionIcon.js (hunk 2 above)
Comment (IvanKryzhanovskyi) [reply src/components/ProhibitionIcon/ProhibitionIcon.js]: Done

File: src/components/ProhibitionIcon/ProhibitionIcon.js (hunk 4 above)
Comment (IvanKryzhanovskyi) [reply src/components/ProhibitionIcon/ProhibitionIcon.js]: Done

File: src/components/ProhibitionIcon/ProhibitionIcon.js (hunk 3 above)
Comment (IvanKryzhanovskyi) [reply src/components/ProhibitionIcon/ProhibitionIcon.js:16]: According to A-M comment we should update text

**Review by IhorKoba — APPROVED** [review]

**Review by mariia-aloshyna — APPROVED** [review]

**Pull request merged**